api/net/Areas/Subscriber/Controllers/ReportController.cs

```diff
@@ -0,0 +1,225 @@
+    private readonly JsonSerializerOptions _serializerOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a ReportController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="reportService"></param>
+    /// <param name="userService"></param>
+    /// <param name="reportHelper"></param>
+    /// <param name="serializerOptions"></param>
+    public ReportController(
+        IReportService reportService,
+        IUserService userService,
+        IReportHelper reportHelper,
+        IOptions<JsonSerializerOptions> serializerOptions)
+    {
+        _reportService = reportService;
+        _userService = userService;
+        _reportHelper = reportHelper;
+        _serializerOptions = serializerOptions.Value;
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Find an array of report for the specified query filter.
+    /// Only returns reports owned by the current user.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<ReportModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Report" })]
+    public IActionResult Find()
+    {
+        var uri = new Uri(this.Request.GetDisplayUrl());
+        var query = Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery(uri.Query);
+        var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
+        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
+        var filter = new ReportFilter(query)
+        {
+            OwnerId = user.Id
+        };
+        return new JsonResult(_reportService.Find(filter).Select(ds => new ReportModel(ds, _serializerOptions)));
+    }
+
+    /// <summary>
+    /// Find all "my" reports.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("my-reports")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<ReportModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Report" })]
+    public IActionResult FindMyFolders()
+    {
+        var username = User.GetUsername() ?? throw new NotAuthorizedException("Username is missing");
+        var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
+        return new JsonResult(_reportService.FindMyReports(user.Id).Select(ds => new ReportModel(ds, _serializerOptions)));
+    }
+
+    /// <summary>
+    /// Find all "available" reports.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("all")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IEnumerable<ReportModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "Report" })]
+    public IActionResult FindAll()
```

Comment:
I think you are trying to use this endpoint for Public reports.  Rename to `GetPublicReports`